### PR TITLE
8610 sync delete records

### LINKED ISSUE #8610: Sync Delete records
- [ ] Purchase Order
- [ ] Purchase Order Line
- [ ] Goods Received
- [x] Goods Received Line

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #8610

# 👩🏻‍💻 What does this PR do?
Sync deletes for pull / push: 
- Purchase orders & lines
- Goods received & lines

There's a weird side affect going on whereas if you delete the line, the costs in the purchase order has changed? I'm told that this is done in a view... but erm would suggest a change in architecture here @andreievg 

# 🧪 Testing

- [ ] Create some purchase orders in OMS
- [ ] Sync
- [ ] Delete
- [ ] Sync
- [ ] Check OG
- [ ] Do same with the lines / goods received

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

# 📃 Reviewer Checklist

The PR Reviewer(s) should fill out this section before approving the PR

**Breaking Changes**
- [ ] No Breaking Changes in the Graphql API
- [ ] Technically some Breaking Changes but not expected to impact any integrations

**Issue Review**
- [ ] All requirements in original issue have been covered
- [ ] A follow up issue(s) have been created to cover additional requirements

**Tests Pass**
- [ ] Postgres
- [ ] SQLite
- [ ] Frontend

### CODE REVIEW

File: server/repository/src/db_diesel/goods_received_line_row.rs [hunk 1]
```diff
@@ -143,8 +143,8 @@ impl Upsert for GoodsReceivedLineRow {
 }
 
 #[derive(Debug)]
-pub struct GoodsReceivedLineRowDelete(pub String);
-impl Delete for GoodsReceivedLineRowDelete {
+pub struct GoodsReceivedLineDelete(pub String);
```
Comment (roxy-dao) [inline server/repository/src/db_diesel/goods_received_line_row.rs:146]: Sorry just nit

File: server/repository/src/db_diesel/goods_received_row.rs [hunk 2]
```diff
@@ -1,12 +1,11 @@
-use crate::Delete;
 use crate::{
-    ChangeLogInsertRow, ChangelogRepository, ChangelogTableName, RepositoryError, RowActionType,
-    StorageConnection, Upsert,
+    ChangeLogInsertRow, ChangelogRepository, ChangelogTableName, Delete, RepositoryError,
```
Comment (roxy-dao) [inline server/repository/src/db_diesel/goods_received_row.rs:2]: more nits throughout...

File: server/service/src/sync/translations/purchase_order.rs [hunk 3]
```diff
@@ -57,69 +56,6 @@ pub struct PurchaseOrderOmsFields {
     pub status: PurchaseOrderStatus,
 }
 
-/** Example record
```
Comment (roxy-dao) [inline server/service/src/sync/translations/purchase_order.rs:60]: Deleted.. this should already be in test_data...

File: server/service/src/sync/test/pull_and_push.rs [hunk 4]
```diff
@@ -120,17 +120,18 @@ async fn test_sync_pull_and_push() {
         .map(|r| (r.record_id.clone(), r.table_name.clone()))
         .collect::<Vec<(String, String)>>();
 
-    let mut i = 0;
-    for translated_record in changelog_translated_records_id_and_table_name.clone() {
+    for (i, translated_record) in changelog_translated_records_id_and_table_name
```
Comment (roxy-dao) [inline server/service/src/sync/test/pull_and_push.rs:123]: Side quest... just a tiny refactor... enumerate already does what we want, so no need for a mut i

File: server/service/src/sync/test/test_data/goods_received.rs [hunk 5]
```diff
@@ -30,8 +28,28 @@ const GOODS_RECEIVED: (&str, &str) = (
...
+    "3486239A597646B2B7259D91A24988E9",
+    r#"{
+        "ID": "3486239A597646B2B7259D91A24988E9",
+        "budget_ID": "",
+        "comment": "",
+        "donor_id": "",
+        "entry_date": "2025-07-24",
+        "linked_transaction_ID": "",
+        "purchase_order_ID": "",
+        "received_date": "2025-07-24",
```
Comment (fergie-nz) [inline server/service/src/sync/test/test_data/goods_received.rs:41]: I think we did the deserialisers wrong - and doesn't handle empty string (null) optional dates.

Would you be willing to change received_date deserialiser to this?

I found this on my manual testing - create a goods_received, sync to OG, fails to sync back with empty date.

```
    #[serde(deserialize_with = "zero_date_as_option")]
    #[serde(serialize_with = "date_option_to_isostring")]
    pub received_date: Option<NaiveDate>,
```

**Review by fergie-nz — APPROVED** [review]
Nice looks great, thank you Roxy.

Just one comment on faulty deserialiser (not related to this PR) - but maybe easy to add here and cover with changing a test case to a null date coming back from OG?
